### PR TITLE
Delete Item button

### PR DESCRIPTION
**Pull request by ldbonkowski**
### Describe your changes

This PR accomplishes the following:

- Adds a button to delete an item

### Issue ticket number and link

This PR addresses: [#16 ](https://github.com/CMPUT301F23T08/HouseHomey/issues/16)

### Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] Necessary java docs are present
- [x] Necessary UML diagrams are created
- [ ] Necessary tests are written

### Screenshots (if applicable)

e.g. UI screenshots, UML screenshots, etc

### CODE REVIEW

File: app/src/main/java/com/example/househomey/HomeFragment.java [hunk 1]
```diff
@@ -44,10 +44,8 @@ public class HomeFragment extends Fragment implements FilterCallback {
 
     /**
      * This constructs a new HomeFragment with the appropriate list of items
-     * @param itemRef A reference to the firestore collection containing the items to display
      */
-    public HomeFragment(CollectionReference itemRef) {
-        this.itemRef = itemRef;
+    public HomeFragment() {
```
Comment (owencooke) [inline app/src/main/java/com/example/househomey/HomeFragment.java:48]: If the Constructors for fragments are bad, let's just move this line to the beginning of onCreateView and get rid of the constructor.

File: app/src/main/java/com/example/househomey/ViewItemFragment.java [hunk 2]
```diff
@@ -57,11 +61,19 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
...
         );
 
+        rootView.findViewById(R.id.delete_button).setOnClickListener(v ->
+                {
+                    ((MainActivity) requireActivity()).getItemRef().document(item.getId()).delete();
+                    navigateToFragmentPage(getContext(), new HomeFragment());
+                }
+        );
+
+        // Do I pass the itemRef to this fragment or just query the db directly
```
Comment (owencooke) [inline app/src/main/java/com/example/househomey/ViewItemFragment.java:75]: lol remove active thoughts

**Review by owencooke — CHANGES_REQUESTED** [review]
Sweet and simple. Just minor changes

**Review by jdrco — APPROVED** [review]
LGTM

**Review by owencooke — APPROVED** [review]
🚀

**Pull request merged**